securedrop/template_filters.py

```diff
@@ -49,3 +49,15 @@ def filesizeformat(value: int) -> str:
         prefix = prefixes[i]
         bytes = float(value) / base ** (i + 1)
         return units.format_unit(bytes, prefix, locale=locale, length="short")
+
+
+def html_datetime_format(dt: datetime) -> str:
+    """
+    Return a datetime string that will pass HTML validation and be
+    adjusted for the application locale
+    """
+    return dates.format_datetime(
+        dt,
+        'yyyy-MM-dd HH:mm:ss.SSS',
+        locale=get_locale()
```

Comment:
Thanks for adding this.  @zenmonkeykstop and I chatted and realized that the `locale` argument here is superfluous, since however the timestamp is localized will be overridden by the explicit pattern required for the `time[datetime]` attribute.  What needs localizing is rather the `time[title]` attribute—which I'll file as a separate issue.

Even if we don't need the `locale`, however, it's nice to have this reusable in a Jinja filter.  :-)  Thanks for taking the time to refactor!

Reply:
Thanks for your time, @cfm, just to clarify- are you suggesting that you'd like to:

1. Leave the code as is and squash ready for merge, despite `locale` not being required? or
2. Remove the `locale` argument from the code but keep the custom filter and squash ready for merge

Not quite sure if you are suggesting that it's nice to have the `locale`, or nice to have the filter, without the `locale`. :)

Reply:
Ah, yes: (2), please.  I meant to acknowledge that, even though the `locale` argument was the original motivation for asking you to refactor from `strftime` to a filter, it still makes sense to keep this date format in filter form along with the others.  Thanks for clarifying.  :-)